app/services/ccms/case_add_requestor.rb

```diff
@@ -133,7 +133,7 @@ def generate_client(xml)
 
     def generate_provider_details(xml)
       xml.__send__('ns2:ProviderCaseReferenceNumber', 'PC4') # TODO: insert @legal_aid_application.provider_case_reference_number when it is available in Apply
-      xml.__send__('ns2:ProviderFirmID', provider.firm.id)
+      xml.__send__('ns2:ProviderFirmID', provider.firm_id)
       xml.__send__('ns2:ProviderOfficeID', provider.selected_office_id)
```

Comment:
Firm id: As Mark said, the firm_id is the UUID of the record on our system.  What you need here is the  `ccms_id` on the `Provider`'s `Firm` record.

Office id:  I beleive that when the user logs in, the selected office_id is set on the `Provider` record, and this is used to populate the `office_id` field on the `LegalAidApplication`.  It's dangerous to use the `Provider#selected_office_id` because when the CCMS payload is generated (as a background task), the provider may have logged out and back in as another office.  Therefore you should use the `ccms_id` of the `LegalAidApplication`'s `Office` record.

Just check that my assumptions here are correct though

Reply:
@stephenrichards `provider.firm.ccms_id` and `@legal_aid_application.office.ccms_id` look good. I'll make the changes as Will is off today and it would be good to get this working so testing can start.